src/main/java/org/ohdsi/webapi/security/model/EntityPermissionSchema.java

```diff
@@ -2,19 +2,27 @@
 
 import org.ohdsi.webapi.model.CommonEntity;
 import org.ohdsi.webapi.shiro.Entities.RoleEntity;
+import org.ohdsi.webapi.shiro.filters.UpdateAccessTokenFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.ohdsi.webapi.shiro.PermissionManager;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 public abstract class EntityPermissionSchema {
+    private final Logger logger = LoggerFactory.getLogger(UpdateAccessTokenFilter.class);
 
     private final EntityType entityType;
     private final Map<String, String> readPermissions;
     private final Map<String, String> writePermissions;
 
```

Comment:
There might actually be two different config options we are thinking of instead of one. The first is the external authorization, where a service checks if someone should have a role. The other would decide if the shared artifact owner/reader roles are used.

Reply:
Not sure if we need a config option for the latter. I think the mere presence of these roles, as part of the roles assigned to the user should trigger the logic.

Reply:
Ok - let's go with that and test it.